apps/extension/server/src/server.ts

```diff
@@ -192,21 +200,42 @@ async function handleDocumentChange(textDocument: TextDocument): Promise<void> {
     }
     const [settings, machineResults] = await Promise.all([
       getDocumentSettings(textDocument.uri),
-      filterOutIgnoredMachines(extracted).machines,
+      filterOutIgnoredMachines(extracted).machines.filter(isMachineResult),
     ]);
 
-    const types = machineResults
-      .filter(
-        (machineResult): machineResult is NonNullable<typeof machineResult> =>
-          !!machineResult?.machineCallResult.definition?.tsTypes?.node,
-      )
-      .map((machineResult, index) =>
-        getTypegenData(
-          UriUtils.basename(URI.parse(textDocument.uri)),
-          index,
-          machineResult,
-        ),
-      );
+    // Iterate over the machine results and add possible typegen data plus config errors
```

Comment:
IMHO this repeats the implementation too much and might have consistency issues with evolving implementation. I'd prefer avoiding comments that don't add additional information to the code or ones that explain particularly complex logic etc.